lib/csv_sniffer.ex

```diff
@@ -64,24 +64,64 @@ defmodule CsvSniffer do
     ~r'(?P<delim>#{@delimiters_regex})(?P<quote>["\']).*?(?P=quote)(?P=delim)'sm,
     #  ".*?",
     ~r'(?:^|\n)(?P<quote>["\']).*?(?P=quote)(?P<delim>#{@delimiters_regex})'sm,
-    ~r'(?:^|\r)(?P<quote>["\']).*?(?P=quote)(?P<delim>#{@delimiters_regex})'sm,
     # ,".*?"
     ~r'(?P<delim>#{@delimiters_regex})(?P<quote>["\']).*?(?P=quote)(?:$|\n)'sm,
-    ~r'(?P<delim>#{@delimiters_regex})(?P<quote>["\']).*?(?P=quote)(?:$|\r)'sm,
     #  ".*?" (no delim)
-    ~r'(?:^|\n)(?P<quote>["\']).*?(?P=quote)(?:$|\n)'sm,
-    ~r'(?:^|\r)(?P<quote>["\']).*?(?P=quote)(?:$|\r)'sm
+    ~r'(?:^|\n)(?P<quote>["\']).*?(?P=quote)(?:$|\n)'sm
   ]
```

Comment:
we remove all the regex expressions with `"\r"`, because it the re-encoding process we already made sure `"\n"` is the only carriage-return character